diff --git a/mem_bkdr_util/mem_bkdr_util.core b/mem_bkdr_util/mem_bkdr_util.core
index 80fcc50..cd99ade 100644
--- a/mem_bkdr_util/mem_bkdr_util.core
+++ b/mem_bkdr_util/mem_bkdr_util.core
@@ -10,12 +10,9 @@ filesets:
     depend:
       - lowrisc:opentitan:bus_params_pkg
       - lowrisc:dv:dv_utils
-      - lowrisc:dv:crypto_dpi_prince:0.1
       - lowrisc:dv:crypto_dpi_present:0.1
       - lowrisc:prim:cipher_pkg:0.1
       - lowrisc:prim:secded:0.1
-      - lowrisc:dv:digestpp_dpi
-      - lowrisc:ip:kmac_pkg
     files:
       - mem_bkdr_util_pkg.sv
       - mem_bkdr_util_row_adapter.sv: {is_include_file: true}
