diff --git a/src/dm_csrs.sv b/src/dm_csrs.sv
index b647c9b..833cdd2 100644
--- a/src/dm_csrs.sv
+++ b/src/dm_csrs.sv
@@ -34,6 +34,7 @@ module dm_csrs #(
   output dm::dmi_resp_t                     dmi_resp_o,
   // global ctrl
   output logic                              ndmreset_o,      // non-debug module reset active-high
+  input  logic                              ndmreset_ack_i,  // non-debug module reset ack pulse
   output logic                              dmactive_o,      // 1 -> debug-module is active,
                                                              // 0 -> synchronous re-set
   // hart status
@@ -513,8 +514,8 @@ module dm_csrs #(
       data_d = data_i;
     end
 
-    // set the havereset flag when we did a ndmreset
-    if (ndmreset_o) begin
+    // set the havereset flag when the ndmreset completed
+    if (ndmreset_ack_i) begin
       havereset_d_aligned[NrHarts-1:0] = '1;
     end
     // -------------
diff --git a/src/dm_top.sv b/src/dm_top.sv
index c21e58d..c357025 100644
--- a/src/dm_top.sv
+++ b/src/dm_top.sv
@@ -32,6 +32,7 @@ module dm_top #(
   input  logic                  rst_ni,
   input  logic                  testmode_i,
   output logic                  ndmreset_o,  // non-debug module reset
+  input  logic                  ndmreset_ack_i, // acknowledement of non-debug module reset
   output logic                  dmactive_o,  // debug module is active
   output logic [NrHarts-1:0]    debug_req_o, // async debug request
   // communicate whether the hart is unavailable (e.g.: power down)
@@ -87,6 +88,8 @@ module dm_top #(
   logic [dm::DataCount-1:0][31:0]   data_mem_csrs;
   logic                             data_valid;
   logic                             ndmreset;
+  logic                             ndmreset_pending;
+  logic                             ndmreset_ack;
   logic [19:0]                      hartsel;
   // System Bus Access Module
   logic [BusWidth-1:0]              sbaddress_csrs_sba;
@@ -123,6 +126,7 @@ module dm_top #(
     .dmi_resp_ready_i,
     .dmi_resp_o,
     .ndmreset_o              ( ndmreset              ),
+    .ndmreset_ack_i          ( ndmreset_ack          ),
     .dmactive_o,
     .hartsel_o               ( hartsel               ),
     .hartinfo_i,
@@ -227,4 +231,20 @@ module dm_top #(
     .rdata_o                 ( slave_rdata_o         )
   );
 
+  always_ff @(posedge clk_i or negedge rst_ni) begin : handle_reset_pending
+    if (!rst_ni) begin
+      ndmreset_pending <= 1'b0;
+    end else begin
+      if (ndmreset) begin
+        ndmreset_pending <= 1'b1;
+      end else if (ndmreset_ack) begin
+        ndmreset_pending <= 1'b0;
+      end
+    end
+  end
+
+  // Gate the acknowledgement input on whether a reset is pending and if there
+  // is not another non-debug module reset being requested immediately.
+  assign ndmreset_ack = ndmreset_pending & !ndmreset & ndmreset_ack_i;
+
 endmodule : dm_top
