diff --git a/core/cache_subsystem/wt_axi_adapter.sv b/core/cache_subsystem/wt_axi_adapter.sv
index 8b2a0c0..4170969 100644
--- a/core/cache_subsystem/wt_axi_adapter.sv
+++ b/core/cache_subsystem/wt_axi_adapter.sv
@@ -551,6 +551,14 @@ module wt_axi_adapter
           end
         end
       end
+      // Fixup for this shift register which assumes all bursted reads are data cache line width.
+      // This is false when the data cache is disabled and CHERI capabilities are 128 bit, bus is 64 bit and data cahce line is 256 bit.
+      if (CVA6Cfg.CheriPresent && axi_rd_last) begin
+        for (int i = 1; i < CVA6Cfg.DCACHE_LINE_WIDTH / CVA6Cfg.AxiDataWidth; i+=2) begin
+          dcache_rd_shift_d[i] = axi_rd_data;
+          dcache_rd_shift_user_d[i] = axi_rd_user;
+        end
+      end
     end else if (CVA6Cfg.RVA && dcache_sc_rtrn) begin
       // encode lr/sc success
       dcache_rd_shift_d[0] = '0;
