diff --git a/core/id_stage.sv b/core/id_stage.sv
index 05f9b77..409e903 100644
--- a/core/id_stage.sv
+++ b/core/id_stage.sv
@@ -472,7 +472,7 @@ module id_stage #(
     if (commit_redirect_q) int_mode_d = int_mode_issue_i;
     else if (mispredict_redirect_i) int_mode_d = int_mode_resolved_branch_i;
     else begin
-      for (int i = 0; i <= CVA6Cfg.NrIssuePorts; i++) begin
+      for (int i = 0; i < CVA6Cfg.NrIssuePorts; i++) begin
         if (fetch_entry_ready_o[i]) begin
           int_mode_d = int_mode_decode_o[i];
         end
